File: DSA/GEHNA_KHANNA_590013140/day23question1.c
#include <stdio.h>
#include <stdlib.h>

struct Node {
    int data;
    struct Node* next;
};

int k_node(struct Node* head, int k) {
    struct Node *first = head, *second = head;
    int i;
    for (i = 0; i < k; i++) {
        if (first == NULL)
            return -1;
        first = first->next;
    }
    while (first != NULL) {
        first = first->next;
        second = second->next;
    }
    if (second != NULL)
        return second->data;
    else
        return -1;
}

struct Node* newNode(int data) {
    struct Node* temp = (struct Node*)malloc(sizeof(struct Node));
    temp->data = data;
    temp->next = NULL;
    return temp;
}

int main() {
    int n, k, value;
    struct Node *head = NULL, *tail = NULL;

    printf("Enter number of nodes: ");
    scanf("%d", &n);

    printf("Enter linked list values:\n");
    for (int i = 0; i < n; i++) {
        scanf("%d", &value);
        struct Node* temp = newNode(value);
        if (head == NULL)
            head = tail = temp;
        else {
            tail->next = temp;
            tail = temp;
        }
    }

    printf("Enter k: ");
    scanf("%d", &k);

    int result = k_node(head, k);
    printf("Output: %d\n", result);

    return 0;
}

File: DSA/Nivedan_590014904/Day_23_Question_1.py
class Node:
    def __init__(self,data,next_node=None):
        self.data =data
        self.next = next_node
n = int(input("Enter the number of nodes: "))
head = None
current = None
for i in range(n):
    val = int(input(f"Enter the data of the {i+1} node: "))
    new_node = Node(val)
    if head is None:
        head = new_node
        current = new_node
    else:
        current.next = new_node
        current = new_node
k = int(input("Enter thr value of k: "))
if k>n:
    print(-1)
else:
    p1 = head
    p2= head
    for j in range(k):
      p1 = p1.next
    while p1 is not None:
      p1 = p1.next
      p2 = p2.next
    print(p2.data)
